python/amici/sbml_import.py

```diff
@@ -1473,6 +1468,40 @@ def _parse_special_functions(sym: sp.Expr, toplevel: bool = True) -> sp.Expr:
     return sym
 
 
+def _parse_piecewise_to_heaviside(args: Iterable[sp.Expr]) -> sp.Expr:
+    """
+   Piecewise functions cannot be transformed into C++ right away, but AMICI has
+   a special interface for Heaviside function, so we transform them
+
+    :param args:
+        symbolic expressions for arguments of the piecewise function
+    """
+    # how many condition-expression pairs will we have?
+    formula = sp.sympify('0')
+    lastroot = sp.Float(1.0)
+    conditions = list(grouper(args, 2, False))
+    for coeff, root in conditions:
+        if root != sp.false:
+            # Sympy doesn't assume by default that the expression is real...
+            root._assumptions['extended_real'] = True
+            # we now need to convert the relational >, >=, ... expression into
+            # a trigger function which will be a Heaviside argument
+            if root.is_Relational:
+                trigger = root.args[0] - root.args[1]
+                if type(root) in (sp.core.relational.StrictLessThan,
+                                  sp.core.relational.LessThan):
+                    trigger = -trigger
+            else:
+                raise SBMLException('AMICI can not parse peicewise functions '
+                                    'with non-relational arguments.')
```

Comment:
Show ID / formula of the problematic function

Reply:
You're right, but I will refactor this anyway now...